### DISCUSSION TITLE
some part of content has been left out

### DISCUSSION THREAD

**Discussion by buptyyf** [pymupdf/PyMuPDF#2266] [Q&A] [answered]
Example pdf:
[动手学深度学习.pdf](https://github.com/pymupdf/PyMuPDF/files/10838959/default.pdf)

<img width="1012" alt="image" src="https://user-images.githubusercontent.com/13342539/221548072-e1a44561-9c2a-4da7-bf69-2638560f15e9.png">

The part of this page can't be parsed by pymupdf. 

```
doc = fitz.open('xxx.pdf')
doc[27].get_text('dict')
```
All blocks are text type. Maybe this part is text, but its text info is also not included in blocks.
<img width="1132" alt="image" src="https://user-images.githubusercontent.com/13342539/221551794-577266e1-e6e5-4e8a-b294-80e2d60e98cd.png">

**Comment by JorjMcKie** [comment, marked as answer]
This is a "Discussions" item, so I first transfer it.

**Reply by JorjMcKie** [reply]
The objects inside the red rectangle are no text - nothing has been left out.
What you see there is vector graphics.
Confirm this by trying to mark anything inside that box using some PDF reader.

**Reply by buptyyf** [reply]
Can vector graphics be parsed by any API in pymupdf? How can I get it?

**Reply by JorjMcKie** [reply]
> Can vector graphics be parsed by any API in pymupdf? How can I get it?

Look at the description of `page.get_drawings()`. This extracts vector graphics from the page.

**Reply by buptyyf** [reply]
Thanks a lot.
